src/workers/parser/getScopes/visitor.js

```diff
@@ -133,11 +134,19 @@ export function parseSourceScopes(sourceId: SourceId): ?Array<ParsedScope> {
 
   const state = {
     sourceId,
+    freeVariables: new Map(),
     scope: lexical,
     scopeStack: []
   };
   t.traverse(ast, scopeCollectionVisitor, state);
 
+  for (const [key, freeVariables] of state.freeVariables) {
+    const binding = global.bindings[key];
+    if (binding) {
+      binding.refs = freeVariables.concat(binding.refs);
+    }
+  }
```

Comment:
does it also help to have a unit test for the visitor so that we have more direct tests of this case, which would be easier to fix than the mochitest